### PR TITLE
fix: fall back to English if translation key is missing

### LINKED ISSUE #1221: Reserved - translations fixes
**This is reserved for @0xf11nix - do not apply for this bounty**

### Overview

Recently, we added a bunch of non-English translation files.

These work great if there's an associated translation key in the non-English files.

As we're moving quickly and adding a lot of UI, we have a lot of missing keys in other languages.

If a key is missing, the UI is falling back to the translation key (see screenshot below), rather than the English translation.

### References and additional details

Good starting point is https://github.com/shapeshift/web/blob/develop/src/assets/translations/index.ts#L9-L10

Your solution can include either

* check the docs of `react-polyglot`, the translation library we use, to see if it's configurable to fall back to a default language (English) in the absence of a translation key (preferred)
* write a script in `./scripts` that can be run via `yarn translations` to automatically spread missing key/value pairs from English to all the other translations.

### Acceptance Criteria

Switching to a non-English translation will fail gracefully with the English string.

### Need By Date

n/a

### Screenshots/Mockups

image

### Ownership

- [X] If my bounty needs engineering or needs product I have added the respective labels on the right
- [X] As the sponsor of this bounty I will review the changes in a preview environment (ops/product) or review the PR (engineering)

### Bounty Hunters

- [Join our discord](https://discord.gg/shapeshift)
- Include an expected timeline for you to complete work in the work plan when you apply for this bounty!
- Please refer to [this link](https://shapeshiftdao.gitbook.io/getting-started/how-to-contribute/bounties#resources-for-new-contributors-and-bounty-hunters) for some basic info
- Please ***do not*** start work on this issue until you are approved in Gitcoin.

### PR DESCRIPTION
**Pull request by 0xf11nix**
## Description

This PR adds fallback English translations in cases where a specific translation key is missing.
Relevant documentation for the underlying library can be found [here](https://github.com/airbnb/polyglot.js#options-overview).

While I was initially concerned that this approach may cause issues with the pluralization, it does not appear that this codebase is currently making use of that functionality anyway.
Additionally, it seems that there is already plenty of discussion underway re. [Making the translations fetched, granular and consistent](https://github.com/shapeshift/web/discussions/1063), so I took the min diff approach.

## Notice

- [x] Have you followed the guidelines in our [Contributing]('https://github.com/shapeshift/web/CONTRIBUTING.md) guide?
- [x] Have you checked to ensure there aren't other open [Pull Requests](https://github.com/shapeshift/web/pulls) for the same update/change?

## Pull Request Type

- [x] :bug: Bug fix (Non-breaking Change: Fixes an issue)
- [ ] :hammer_and_wrench: Chore (Non-breaking Change: Doc updates, pkg upgrades, typos, etc..)
- [ ] :nail_care: New Feature (Breaking/Non-breaking Change)

## Issue (if applicable)
Closes #1221 

## Risk

This change could potentially cause some regression with the existing translations, but I would suggest that risk is on the low side. One possibility would be (as mentioned above), if we were to attempt to support pluralization under the current library. This would likely result in unexpected behaviour for some languages, in some circumstances, where a fallback translation is used.

## Testing

My intention was to include at least some integration/regression tests for these changes, though I am conscious that the changes, as they stand, are already going to conflict with #1216, and perhaps abstracting the provider (similarly to what was done in 1216) for the ease of testing might be out of scope for this change. 
@0xdef1cafe If you think it makes sense, or would prefer an alternative approach for testing the missing translations, I am happy to make the changes and include them in this PR.

### CODE REVIEW

File: src/AppProviders.tsx [hunk 1]
```diff
@@ -21,6 +22,9 @@ import { theme } from 'theme/theme'
 
 const locale: string = simpleLocale()
 const messages = translations[locale]
+const onMissingKey = (key: string) => {
+  return get(translations['en'], key)
+}
```
Comment (0xdef1cafe) [inline src/AppProviders.tsx:27]:
```suggestion
const onMissingKey = (key: string) => get(translations['en'], key)
```

File: src/AppProviders.tsx [hunk 2]
```diff
@@ -1,5 +1,6 @@
 import { ChakraProvider, ColorModeScript } from '@chakra-ui/react'
 import { DefiManagerProvider } from 'features/defi/contexts/DefiManagerProvider/DefiManagerProvider'
+import { get } from 'lodash'
```
Comment (0xdef1cafe) [inline src/AppProviders.tsx:3]:
```suggestion
import get from 'lodash/get'
```

this will tree shake better

File: src/AppProviders.tsx (hunk 1 above)
Comment (0xdef1cafe) [reply src/AppProviders.tsx:27]: me being a stylistic nit

**Comment by 0xApotheosis** [comment]
I'm not sure why the checks didn't run 🤔 

@0xf11nix is it worth adding a small Cypress test to check the `connect-wallet` screen correctly falls back when a language doesn't exist?

Cypress hasn't been used to check internationalization yet though, so we have no existing patterns, and it might not be as simple as it sounds.

Otherwise LGTM!

**Comment by 0xdef1cafe** [comment]
> 

github was rooted again this morning, should run when he pushes another commit.

@0xf11nix as discussed via discord, it's ok without tests here, as we can prove it functionally in the preview environment, and this fix is better than the current state.

**Comment by 0xf11nix** [comment]
@0xdef1cafe Seems like the tests still aren't running. I force-pushed my rebase, but I would have expected that to trigger a rebuild. Is there something I can do to give it the poke it's waiting for?

**Comment by gomesalexandre** [comment]
> @0xdef1cafe Seems like the tests still aren't running. I force-pushed my rebase, but I would have expected that to trigger a rebuild. Is there something I can do to give it the poke it's waiting for?

@0xf11nix Try this: `git commit --allow-empty -m "chore: trigger CI`

**Review by theoboldfrazier — APPROVED** [review]

**Review by 0xApotheosis — APPROVED** [review]

**Comment by NeOMakinG** [comment]
@0xf11nix There is a small conflict that you need to resolve!

**Review by NeOMakinG — APPROVED** [review]

**Review by 0xdef1cafe — APPROVED** [review]

**Pull request merged**